docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
+
+![allow-egress-to-dns](../../images/allow-egress-to-dns.png)
+
+* Always allow ingress traffic from monitoring namespace to all workloads
+
+![allow-ingress-from-monitoring](../../images/allow-ingress-from-monitoring.png)
+
+* Delegate to namespace scoped network policies (we will learn more about what this is
+  in the following sections)
+
+![delegate-sample](../../images/delegate-sample.png)
+
+* Always deny traffic to sensitive namespaces from everywhere
+
+![sensitive-deny](../../images/sensitive-deny.png)
+
+## How to enable this feature on an OVN-Kubernetes cluster?
+
+There is a Feature Config option `enable-admin-network-policy` under
+`OVNKubernetesFeatureConfig` config that can be used to enable or disable this
+feature. When the feature is enabled, the CRDs are installed on the cluster
+and the `AdminNetworkPolicy` controller is invoked.
+
+```shell
+$ oc get crd adminnetworkpolicies.policy.networking.k8s.io
+NAME                                            CREATED AT
+adminnetworkpolicies.policy.networking.k8s.io   2023-10-01T11:33:58Z
+$ oc get crd baselineadminnetworkpolicies.policy.networking.k8s.io
+NAME                                                    CREATED AT
+baselineadminnetworkpolicies.policy.networking.k8s.io   2023-10-01T11:33:58Z
 ```
+
+## Workflow Description
+
+## Implementation Details
+
+### User facing API Changes
+
+Like mentioned above, the API lives in the
+[kubernetes-sigs](https://github.com/kubernetes-sigs/network-policy-api) repo.
+
+#### AdminNetworkPolicy Sample API
+
+Let us look at a sample ANP yaml:
+
+```yaml
 apiVersion: policy.networking.k8s.io/v1alpha1
 kind: AdminNetworkPolicy
 metadata:
-  name: pass-example
+  name: cluster-control
 spec:
-  priority: 10
+  priority: 34
   subject:
-    namespaces:
-      matchLabels:
-          conformance-house: gryffindor
+    namespaces: {}
   ingress:
-  - name: "deny-all-ingress-from-slytherin"
-    action: "Deny"
+  - name: "allow-from-ingress-nginx" # rule0
+    action: "Allow"
     from:
     - namespaces:
-        namespaceSelector:
-          matchLabels:
-            conformance-house: slytherin
-  egress:
-  - name: "deny-all-egress-to-slytherin"
+        matchLabels:
+          kubernetes.io/metadata.name: ingress-nginx
+  - name: "allow-from-monitoring" # rule1
+    action: "Allow"
+    from:
+    - namespaces:
+        matchLabels:
+          kubernetes.io/metadata.name: monitoring
+    ports:
+    - portNumber:
+        protocol: TCP
+        port: 7564
+    - namedPort: "scrape"
+  - name: "allow-from-open-tenants" # rule2
+    action: "Allow"
+    from:
+    - namespaces: # open tenants
+        matchLabels:
+          tenant: open
+  - name: "pass-from-restricted-tenants" # rule3
+    action: "Pass"
+    from:
+    - namespaces: # restricted tenants
+        matchLabels:
+          tenant: restricted
+  - name: "default-deny" # rule4
     action: "Deny"
+    from:
+    - namespaces: {}
```

Comment:
yeah this always trips me..

Reply:
`    * rule4: deny all other ingress traffic` wrote it that way

and that is indeed a problem because we don't have a good way to say "allow from yourself"